### PR TITLE
Make run_tests.py python-version agnostic

### PR DESCRIPTION
**Pull request by gnossen**
A few people have noticed that `run_tests.py` cannot be run under python 3. In fact, many files under the `tools/` directory use python2-only features/syntax. This PR makes all python dependencies of `run_tests.py` python2/3 compatible.

```bash
  python3 tools/run_tests/run_tests.py --language python
  File "tools/run_tests/run_tests.py", line 1722
    except subprocess.CalledProcessError, e:
                                        ^
  SyntaxError: invalid syntax
```

### CODE REVIEW

**Comment by grpc-testing** [comment]
```
****************************************************************

libgrpc.so

     VM SIZE        FILE SIZE
 ++++++++++++++  ++++++++++++++

  [ = ]       0        0  [ = ]


****************************************************************

libgrpc++.so

     VM SIZE        FILE SIZE
 ++++++++++++++  ++++++++++++++

  [ = ]       0        0  [ = ]



```

**Comment by grpc-testing** [comment]
```
[trickle] No significant performance differences
```

**Comment by grpc-testing** [comment]
```
[trickle] No significant performance differences
```

**Comment by grpc-testing** [comment]
```
Objective-C binary sizes
*****************STATIC******************
  New size                      Old size
 2,016,509      Total (=)      2,016,509

 No significant differences in binary sizes

***************FRAMEWORKS****************
  New size                      Old size
11,123,085      Total (>)     11,123,078

 No significant differences in binary sizes


```

**Comment by grpc-testing** [comment]
```
Objective-C binary sizes
*****************STATIC******************
  New size                      Old size
 2,016,509      Total (=)      2,016,509

 No significant differences in binary sizes

***************FRAMEWORKS****************
  New size                      Old size
11,123,083      Total (>)     11,123,082

 No significant differences in binary sizes


```

**Comment by grpc-testing** [comment]
```
Corrupt JSON data (indicates timeout or crash): 
    bm_call_create.BM_IsolatedFilter_ClientChannelFilter_NoOp_.counters.new: 10
    bm_call_create.BM_IsolatedFilter_ClientChannelFilter_NoOp_.counters.old: 10


[microbenchmarks] No significant performance differences
```

**Comment by grpc-testing** [comment]
```
****************************************************************

libgrpc.so

     VM SIZE        FILE SIZE
 ++++++++++++++  ++++++++++++++

  [ = ]       0        0  [ = ]


****************************************************************

libgrpc++.so

     VM SIZE        FILE SIZE
 ++++++++++++++  ++++++++++++++

  [ = ]       0        0  [ = ]



```

**Comment by grpc-testing** [comment]
```
[trickle] No significant performance differences
```

**Comment by grpc-testing** [comment]
```
Objective-C binary sizes
*****************STATIC******************
  New size                      Old size
 2,016,505      Total (=)      2,016,505

 No significant differences in binary sizes

***************FRAMEWORKS****************
  New size                      Old size
11,123,077      Total (<)     11,123,080

 No significant differences in binary sizes


```

**Comment by grpc-testing** [comment]
```
****************************************************************

libgrpc.so

     VM SIZE        FILE SIZE
 ++++++++++++++  ++++++++++++++

  [ = ]       0        0  [ = ]


****************************************************************

libgrpc++.so

     VM SIZE        FILE SIZE
 ++++++++++++++  ++++++++++++++

  [ = ]       0        0  [ = ]



```

**Comment by grpc-testing** [comment]
```
[trickle] No significant performance differences
```

**Comment by grpc-testing** [comment]
```
Corrupt JSON data (indicates timeout or crash): 
    bm_call_create.BM_IsolatedFilter_ClientChannelFilter_NoOp_.counters.new: 10
    bm_call_create.BM_IsolatedFilter_ClientChannelFilter_NoOp_.counters.old: 10


[microbenchmarks] No significant performance differences
```

**Comment by grpc-testing** [comment]
```
Corrupt JSON data (indicates timeout or crash): 
    bm_call_create.BM_IsolatedFilter_ClientChannelFilter_NoOp_.counters.new: 10
    bm_call_create.BM_IsolatedFilter_ClientChannelFilter_NoOp_.counters.old: 10


[microbenchmarks] No significant performance differences
```

**Comment by grpc-testing** [comment]
```
****************************************************************

libgrpc.so

     VM SIZE        FILE SIZE
 ++++++++++++++  ++++++++++++++

  [ = ]       0        0  [ = ]


****************************************************************

libgrpc++.so

     VM SIZE        FILE SIZE
 ++++++++++++++  ++++++++++++++

  [ = ]       0        0  [ = ]



```

**Comment by grpc-testing** [comment]
```
[trickle] No significant performance differences
```

**Comment by grpc-testing** [comment]
```
Objective-C binary sizes
*****************STATIC******************
  New size                      Old size
 2,020,616      Total (=)      2,020,616

 No significant differences in binary sizes

***************FRAMEWORKS****************
  New size                      Old size
11,180,938      Total (<)     11,180,947

 No significant differences in binary sizes


```

**Comment by grpc-testing** [comment]
```
Corrupt JSON data (indicates timeout or crash): 
    bm_call_create.BM_IsolatedFilter_ClientChannelFilter_NoOp_.counters.new: 10
    bm_call_create.BM_IsolatedFilter_ClientChannelFilter_NoOp_.counters.old: 10


[microbenchmarks] No significant performance differences
```

**Comment by gnossen** [comment]
Looks like we hit https://github.com/grpc/grpc/issues/16497: 

```
/Volumes/BuildData/tmpfs/src/github/grpc/workspace_objc_macos_opt_native/src/objective-c/tests/InteropTests.m:416: error: -[InteropTestsLocalCleartext testCancelAfterBeginRPC] : ((call.state) equal to (GRXWriterStateFinished)) failed: ("1") is not equal to ("3")
Test Case '-[InteropTestsLocalCleartext testCancelAfterBeginRPC]' failed (0.005 seconds).
```

**Comment by grpc-testing** [comment]
```
****************************************************************

libgrpc.so

     VM SIZE        FILE SIZE
 ++++++++++++++  ++++++++++++++

  [ = ]       0        0  [ = ]


****************************************************************

libgrpc++.so

     VM SIZE        FILE SIZE
 ++++++++++++++  ++++++++++++++

  [ = ]       0        0  [ = ]



```

**Comment by grpc-testing** [comment]
```
[trickle] No significant performance differences
```

**Comment by grpc-testing** [comment]
```
Objective-C binary sizes
*****************STATIC******************
  New size                      Old size
 2,020,616      Total (=)      2,020,616

 No significant differences in binary sizes

***************FRAMEWORKS****************
  New size                      Old size
11,180,936      Total (<)     11,180,949

 No significant differences in binary sizes


```

**Comment by grpc-testing** [comment]
```
Corrupt JSON data (indicates timeout or crash): 
    bm_call_create.BM_IsolatedFilter_ClientChannelFilter_NoOp_.counters.new: 10
    bm_call_create.BM_IsolatedFilter_ClientChannelFilter_NoOp_.counters.old: 10


[microbenchmarks] No significant performance differences
```

File: tools/run_tests/run_tests.py [hunk 1]
```diff
@@ -14,8 +14,6 @@
 # limitations under the License.
 """Run tests in parallel."""
 
-from __future__ import print_function
```
Comment (lidizheng) [inline tools/run_tests/run_tests.py]: We still need this line for Python 2, in case of printing multiple object like line 451.

Comment (gnossen) [reply tools/run_tests/run_tests.py]: Good catch! This line must have gotten pulled out by `2to3`

Comment (gnossen) [reply tools/run_tests/run_tests.py]: @lidizheng Addressed! Ready for re-review.

**Comment by grpc-testing** [comment]
```
****************************************************************

libgrpc.so

     VM SIZE        FILE SIZE
 ++++++++++++++  ++++++++++++++

  [ = ]       0        0  [ = ]


****************************************************************

libgrpc++.so

     VM SIZE        FILE SIZE
 ++++++++++++++  ++++++++++++++

  [ = ]       0        0  [ = ]



```

**Comment by grpc-testing** [comment]
```
[trickle] No significant performance differences
```

**Comment by grpc-testing** [comment]
```
Objective-C binary sizes
*****************STATIC******************
  New size                      Old size
 2,020,616      Total (=)      2,020,616

 No significant differences in binary sizes

***************FRAMEWORKS****************
  New size                      Old size
11,180,945      Total (<)     11,180,947

 No significant differences in binary sizes


```

**Comment by grpc-testing** [comment]
```
[trickle] No significant performance differences
```

**Comment by grpc-testing** [comment]
```
Corrupt JSON data (indicates timeout or crash): 
    bm_call_create.BM_IsolatedFilter_ClientChannelFilter_NoOp_.counters.new: 10
    bm_call_create.BM_IsolatedFilter_ClientChannelFilter_NoOp_.counters.old: 10


[microbenchmarks] No significant performance differences
```

**Pull request merged**